### ISSUE TITLE
Validation errors for HfConfig

### ISSUE THREAD

**Issue by yuwenzho** [microsoft/Olive#847]
**Describe the bug**

When running [open_llama_config.json](https://github.com/microsoft/Olive/blob/main/examples/open_llama/open_llama_config.json), `hf_config` will cause the validation errors:
```json
"input_model":{
        "type": "PyTorchModel",
        "config": {
            "hf_config": {
                "model_name": "openlm-research/open_llama_3b",
                "model_class": "LlamaForCausalLM",
                "components": [{ "name": "decoder_model" }, { "name": "decoder_with_past_model" }]
            }
        }
    }
```

```python
pydantic.v1.error_wrappers.ValidationError: 4 validation errors for HfConfig
components -> 0 -> io_config
  field required (type=value_error.missing)
components -> 0 -> dummy_inputs_func
  field required (type=value_error.missing)
components -> 1 -> io_config
  field required (type=value_error.missing)
components -> 1 -> dummy_inputs_func
  field required (type=value_error.missing)
```

**To Reproduce**

In example of [open_llama](https://github.com/microsoft/Olive/tree/main/examples/open_llama), run `python -m olive.workflows.run --config open_llama_config.json`

**Other information**
 - OS: Linux
 - Olive version: main
 - ONNXRuntime package and version: [e.g. onnxruntime: 1.16.3]

**Additional context**

**Comment by guotuofeng** [comment]
@jambayk, @trajepl, does the io_config/dummy_inputs_func in component need to be mandatory?

**Comment by guotuofeng** [comment]
This is a regression coming from https://github.com/microsoft/Olive/pull/833, we will fix it

**Comment by jambayk** [comment]
@yuwenzho I have created a PR with a potential fix https://github.com/microsoft/Olive/pull/851. Could you try this branch to see if everything works as expected? I tried running the example on my dev machine but the quantization takes too long.

**Comment by guotuofeng** [comment]
> @yuwenzho I have created a PR with a potential fix #851. Could you try this branch to see if everything works as expected? I tried running the example on my dev machine but the quantization takes too long.

could we enable the similar UT?

**Comment by yuwenzho** [comment]
> @yuwenzho I have created a PR with a potential fix #851. Could you try this branch to see if everything works as expected? I tried running the example on my dev machine but the quantization takes too long.

#851 looks good to me. Thanks for the fix.
